spring-cloud-gcp-pubsub-stream-binder/src/test/java/com/google/cloud/spring/stream/binder/pubsub/properties/PubSubExtendedBindingsPropertiesTests.java

```diff
@@ -17,116 +17,103 @@
-    assertThat(binder.getExtendedConsumerProperties("input").isAutoCreateResources()).isTrue();
+  private DefaultBinderFactory createMockExtendedBinderFactory() {
+    BinderTypeRegistry binderTypeRegistry = createMockExtendedBinderTypeRegistry();
+    return new DefaultBinderFactory(
+        Collections.singletonMap("mock",
+            new BinderConfiguration("mock", new HashMap<>(), true, true)),
+        binderTypeRegistry, null);
+  }
 
-    assertThat(binder.getExtendedConsumerProperties("custom-in").getAckMode())
-        .isEqualTo(AckMode.AUTO);
-    assertThat(binder.getExtendedConsumerProperties("input").getAckMode())
-        .isEqualTo(AckMode.AUTO_ACK);
+  private DefaultBinderTypeRegistry createMockExtendedBinderTypeRegistry() {
+    return new DefaultBinderTypeRegistry(
+        Collections.singletonMap("mock", new BinderType("mock",
+            new Class[]{ MockExtendedBinderConfiguration.class })));
   }
 
-  /** Spring Boot config for tests. */
   @Configuration
-  @EnableBinding(PubSubBindingsTestConfiguration.CustomTestSink.class)
-  static class PubSubBindingsTestConfiguration {
-
-    @Bean
-    public PubSubAdmin pubSubAdmin() {
-      PubSubAdmin pubSubAdminMock = Mockito.mock(PubSubAdmin.class);
-      when(pubSubAdminMock.createSubscription(anyString(), anyString()))
-          .thenReturn(Subscription.getDefaultInstance());
-      when(pubSubAdminMock.getSubscription(anyString()))
-          .thenReturn(Subscription.getDefaultInstance());
-      when(pubSubAdminMock.getTopic(anyString())).thenReturn(Topic.getDefaultInstance());
-      return pubSubAdminMock;
-    }
+  public static class MockExtendedBinderConfiguration {
 
+    @SuppressWarnings("rawtypes")
     @Bean
-    public PubSubTemplate pubSubTemplate() {
-      PublisherFactory publisherFactory = Mockito.mock(PublisherFactory.class);
-
-      SubscriberFactory subscriberFactory = Mockito.mock(SubscriberFactory.class);
-      when(subscriberFactory.getProjectId()).thenReturn("test-project");
-      when(subscriberFactory.createSubscriberStub(any()))
-          .thenReturn(Mockito.mock(SubscriberStub.class));
-      when(subscriberFactory.createSubscriber(anyString(), any()))
-          .thenReturn(Mockito.mock(Subscriber.class));
-
-      return new PubSubTemplate(
-          new PubSubPublisherTemplate(publisherFactory),
-          new PubSubSubscriberTemplate(subscriberFactory));
-    }
-
-    @StreamListener("input")
-    public void process(String payload) {
-      System.out.println(payload);
-    }
-
-    @StreamListener("custom-in")
-    public void processCustom(String payload) {
-      System.out.println(payload);
-    }
-
-    /** interface for testing. */
-    interface CustomTestSink extends Sink {
-      @Input("custom-in")
-      SubscribableChannel customIn();
+    public Binder<?, ?, ?> extendedPropertiesBinder() {
+      Binder mock = Mockito.mock(Binder.class,
```

Comment:
minor: How about we make `Mockito` a static import?

Reply:
Done.